modules/loader/src/main/java/com/opengamma/strata/loader/csv/CdsTradeCsvPlugin.java

```diff
@@ -0,0 +1,411 @@
+  private static final String LEGAL_ENTITY_ID_FIELD = "Legal Entity Id";
+  private static final String PAYMENT_ON_DEFAULT_FIELD = "Payment On Default";
+  private static final String PROTECTION_START_FIELD = "Protection Start";
+  private static final String STEP_IN_DATE_OFFSET_DAYS_FIELD = "Step In Date Offset Days";
+  private static final String STEP_IN_DATE_OFFSET_CAL_FIELD = "Step In Date Offset Calendar";
+  private static final String STEP_IN_DATE_OFFSET_ADJ_CNV_FIELD = "Step In Date Offset Adjustment Convention";
+  private static final String STEP_IN_DATE_OFFSET_ADJ_CAL_FIELD = "Step In Date Offset Adjustment Calendar";
+  private static final String SETTLEMENT_DATE_OFFSET_DAYS_FIELD = "Settlement Date Offset Days";
+  private static final String SETTLEMENT_DATE_OFFSET_CAL_FIELD = "Settlement Date Offset Calendar";
+  private static final String SETTLEMENT_DATE_OFFSET_ADJ_CNV_FIELD = "Settlement Date Offset Adjustment Convention";
+  private static final String SETTLEMENT_DATE_OFFSET_ADJ_CAL_FIELD = "Settlement Date Offset Adjustment Calendar";
+
+  //-------------------------------------------------------------------------
+  /**
+   * Parses from the CSV row.
+   * 
+   * @param row  the CSV row
+   * @param info  the trade info
+   * @param resolver  the resolver used to parse additional information
+   * @return the parsed trade
+   */
+  static CdsTrade parse(CsvRow row, TradeInfo info, TradeCsvInfoResolver resolver) {
+    CdsTrade trade = parseRow(row, info, resolver);
+    return resolver.completeTrade(row, trade);
+  }
+
+  // parse the row to a trade
+  private static CdsTrade parseRow(CsvRow row, TradeInfo info, TradeCsvInfoResolver resolver) {
+    BuySell buySell = LoaderUtils.parseBuySell(row.getValue(BUY_SELL_FIELD));
+    double notional = LoaderUtils.parseDouble(row.getValue(NOTIONAL_FIELD));
+    double fixedRate = LoaderUtils.parseDoublePercent(row.getValue(FIXED_RATE_FIELD));
+    String entityScheme = row.findValue(LEGAL_ENTITY_ID_SCHEME_FIELD).orElse(DEFAULT_LEGAL_ENTITY_SCHEME);
+    StandardId entityId = StandardId.of(entityScheme, row.getValue(LEGAL_ENTITY_ID_FIELD));
+    Optional<Tenor> tenorOpt = row.findValue(TENOR_FIELD).map(s -> LoaderUtils.parseTenor(s));
+    Optional<LocalDate> startDateOpt = row.findValue(START_DATE_FIELD).map(s -> LoaderUtils.parseDate(s));
+    Optional<LocalDate> endDateOpt = row.findValue(END_DATE_FIELD).map(s -> LoaderUtils.parseDate(s));
+    Optional<AdjustablePayment> premiumOpt = row.findValue(PREMIUM_DIRECTION_FIELD)
+        .map(ignored -> {
+          CurrencyAmount amount = CsvLoaderUtils.parseCurrencyAmountWithDirection(
+              row, PREMIUM_CURRENCY_FIELD, PREMIUM_AMOUNT_FIELD, PREMIUM_DIRECTION_FIELD);
+          AdjustableDate date = CsvLoaderUtils.parseAdjustableDate(
+              row, PREMIUM_DATE_FIELD, PREMIUM_DATE_CNV_FIELD, PREMIUM_DATE_CAL_FIELD);
+          return AdjustablePayment.of(amount, date);
```

Comment:
Block lambdas back in fashion now? 🤔

Reply:
They have occasional uses, and this demonstrates a use where the lambda is `void` and side-effecting.